client/client_test.go

```diff
@@ -12,67 +12,70 @@ import (
 	"github.com/docker/docker/api"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/internal/testutil"
+	"github.com/gotestyourself/gotestyourself/skip"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestNewEnvClient(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		t.Skip("skipping unix only test for windows")
-	}
-	cases := []struct {
+	skip.IfCondition(t, runtime.GOOS == "windows")
```

Comment:
Why do we need a dependency to mark a skipped test?

Reply:
We don't, but it's better this way. Instead of reporting some meaningless message this prints the condition that returned true, so the literal `runtime.GOOS == "windows"`. Additional messages are also supported, but generally unnecessary.

Reply:
This is a waste of dependency space. The skip message is sufficient.

Reply:
I disagree. Also, this dependency is not being added in this PR, so it's not really relevant.